tensorflow/compiler/xla/service/llvm_ir/llvm_target_helper.h

```diff
@@ -0,0 +1,50 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_XLA_SERVICE_LLVM_IR_LLVM_TARGET_HELPER_H_
+#define TENSORFLOW_COMPILER_XLA_SERVICE_LLVM_IR_LLVM_TARGET_HELPER_H_
+
+#include <string>
+#include "llvm/ADT/Triple.h"
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/Module.h"
+
+namespace xla {
+namespace llvm_ir {
+// Enmerations to get target specific intrinsics or function calls
+enum TargetIntrinsicID {
```

Comment:
If we can't get rid of kLastId, I think you want this to be an enum class, otherwise we're pulling the identifier `kLastId` into namespace `xla::llvm_ir`, which is pretty ambiguous, that could mean basically anything.

Thankfully I don't think you really need kLastId at all.  It's just there to do bounds checking, but I think it kind of gives a false sense of security.  For example, in the current implementation, one could pass a negative number and you'd happily get an out-of-bounds read on the array.

One could write a `switch` with a `default` case, but I'd actually say it's probably better not to, because out-of-bounds enum values are already UB (well, technically it's UB if the value has bits set that aren't set in any enumerator), and if we get rid of the `default` case then we get a compile error if we leave out any enumerator from the switch.